lib/src/main/java/graphql/nadel/engine/transform/hydration/batch/NadelNewBatchHydrationInputBuilder.kt

```diff
@@ -27,22 +51,28 @@ internal object NadelNewBatchHydrationInputBuilder {
         instruction: NadelBatchHydrationFieldInstruction,
         hydrationField: ExecutableNormalizedField,
         sourceIds: List<JsonNode>,
-    ): List<Map<NadelHydrationActorInputDef, NormalizedInputValue>> {
+    ): List<NadelHydrationArgumentsBatch> {
```

Comment:
Created an object to store the original type + the `sourceIds` that made up the batch.